compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -459,7 +451,7 @@ impl<'a, 'b, 'tcx> TypeFolder<'tcx> for AssocTypeNormalizer<'a, 'b, 'tcx> {
                 )
                 .ok()
                 .flatten()
-                .unwrap_or_else(|| ty);
+                .unwrap_or_else(|| ty.super_fold_with(self));
 
                 let normalized_ty = PlaceholderReplacer::replace_placeholders(
```

Comment:
This is a no-op in the "fallback" case, right? Maybe we want

```rust
.ok()
.flatten()
.map(|ty| PlaceholderReplacer::replace_placeholders(...))
.unwrap_or_else(|| ty.super_fold_with(self))
```

?

Reply:
Yes, I'll change it.